packages/react-instantsearch-hooks-dom/src/widgets/__tests__/Highlight.test.tsx

```diff
@@ -0,0 +1,283 @@
+import { render } from '@testing-library/react';
+import React from 'react';
+
+import { Highlight } from '../Highlight';
+
+describe('Highlight', () => {
+  test('renders single match', () => {
+    const { container } = render(
+      <Highlight
+        hit={{
+          objectID: '1',
+          __position: 1,
+          data: 'test',
+          _highlightResult: {
+            data: {
+              matchedWords: ['test'],
+              matchLevel: 'partial',
+              value: '<mark>te</mark>st',
+            },
+          },
+        }}
+        attribute="data"
+      />
+    );
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <span
+          class="ais-Highlight"
+        >
+          <mark
+            class="ais-Highlight-highlighted"
+          >
+            te
+          </mark>
+          <span
+            class="ais-Highlight-nonHighlighted"
+          >
+            st
+          </span>
+        </span>
+      </div>
+    `);
+  });
+
+  test('renders list of matches', () => {
+    const { container } = render(
+      <Highlight
+        hit={{
+          objectID: '1',
+          __position: 1,
+          data: ['test', 'nothing'],
+          _highlightResult: {
+            data: [
+              {
+                matchedWords: ['test'],
+                matchLevel: 'partial',
+                value: '<mark>te</mark>st',
+              },
+              { matchedWords: [], matchLevel: 'none', value: 'nothing' },
+            ],
+          },
+        }}
+        attribute="data"
+      />
+    );
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <span
+          class="ais-Highlight"
+        >
+          <mark
+            class="ais-Highlight-highlighted"
+          >
+            te
+          </mark>
+          <span
+            class="ais-Highlight-nonHighlighted"
+          >
+            st
+          </span>
+          <span
+            class="ais-Highlight-separator"
+          >
+            , 
+          </span>
+          <span
+            class="ais-Highlight-nonHighlighted"
+          >
+            nothing
+          </span>
+        </span>
+      </div>
+    `);
+  });
+
+  test('renders path to match', () => {
+    const { container } = render(
+      <Highlight
+        hit={{
+          objectID: '1',
+          __position: 1,
+          data: { subdata: 'test' },
+          _highlightResult: {
+            data: {
+              subdata: {
+                matchedWords: ['test'],
+                matchLevel: 'partial',
+                value: '<mark>te</mark>st',
+              },
+            },
+          },
+        }}
+        // @ts-expect-error TypeScript type does not accept string, as a way to discourage it
```

Comment:
👍 Did we already deprecate this notation?

Reply:
We don't officially deprecate it. Should we, or do I change the TS type to allow any string, not just top-level keys?

Reply:
I tried a lot to make it work in TS, but it's simply super hard to type it correct, the execution keeps timing out. See https://github.com/algolia/react-instantsearch/pull/3377 for my trial. I don't think we need to deprecate, but we can still leave TypeScript with the "simpler" type

Reply:
Simpler type being that it errors with strings, right? This is a fine tradeoff to me, as long as we don't document the dotted notation but the alternative instead.